indico/modules/events/registration/client/js/form/FormItem.jsx

```diff
@@ -7,6 +7,7 @@
 
 import PropTypes from 'prop-types';
 import React from 'react';
+import {useForm, useFormState} from 'react-final-form';
```

Comment:
(commenting in a random place just to have a separate thread)

One other thing we should think about is manager-only sections. Can you reference a m-only field from a normal field? Can a m-only field reference another m-only field? Can m-only field reference a normal field?

Reply:
> Can you reference a m-only field from a normal field?

Does not make much sense IMHO since data for those fields is not exposed to registrants.

> Can a m-only field reference another m-only field?

Sure, why not?

> Can m-only field reference a normal field?

Not sure if very useful, considering managers can already ignore the "required" flag of fields anyway, so making any manager-only field required (but then conditional) is a bit pointless...

Reply:
> Not sure if very useful

Yeah some of the combinations don't make much sense, the question was more about which ones the UI should allow/disallow

Reply:
While I see the `is_manager_only` field in the `form_items` (which includes both sections and fields), I only see it applicable in sections, but never in fields. Actually the `FormItem` component does not include it, while the `FormSection` one does. Can fields actually be exclusively for managers?

Reply:
Yes, manager-only is a flag set exclusively on the section level. So either a whole section is normal or manager-only.

Reply:
Then things like this one don't make much sense, isn't it?

> Can a m-only field reference another m-only field?

Reply:
I would be perfectly fine with disabling conditionals within manager-only sections. OTOH, it doesn't add much either, and maybe some managers want to use it for better UX for them.

In any case, we'll need a check when editing the section to not allow changing the type when there's a cross-section conditional field involving that section.

Reply:
In [0c1162e](https://github.com/indico/indico/pull/6678/commits/0c1162ea36ce10d7dd90b60c9b4234efe33dadfd) I implemented the logic for not allowing fields to depend on fields contained in management sections.

Reply:
And [ce24ad4](https://github.com/indico/indico/pull/6678/commits/ce24ad47ca41a616d4051ecc63ddcc5a7db08d72) forbids enabling management-only sections if there are conditional fields involved in both cases:
- Fields inside the section being modified.
- External fields from other sections.